dynamicWallpaper.sh

```diff
@@ -67,23 +68,38 @@ set_pape () {
         pape=$(find "$pape_prefix"/"$t_type"/* | shuf -n 1)
     fi
 
+
     if which osascript; then
+        PAPE=$(find  "$pape_prefix"/"$t_type"/"$w_type"/* "$pape_prefix"/"$t_type"/Misc/* | shuf -n 1)
         cmd_str="tell application \"System Events\" to tell every desktop to set picture to \"$pape\""
         osascript -e "$cmd_str"
     else
-        /usr/bin/feh --randomize --bg-fill "$pape"
+        pape=$(find  "$pape_prefix"/"$t_type"/"$w_type"/* "$pape_prefix"/"$t_type"/Misc/* | shuf -n 1)
+	#if [[ $XDG_CURRENT_DESKTOP == XFCE ]]; then
+	if [[ $(w | grep xfce | grep -v grep) ]];then
```

Comment:
FYI you can skip the `grep needle | grep -v grep` thing like this:  `grep [n]eedle`.  i.e. `w | grep [x]fce`

Looking ahead, you might want to take this opportunity to make this a `case` statement to future-proof for other methods e.g.

```
case "$(w)" in
  (*xfce*)
    ...
  ;;
  (*cinnamon*)
    # You might find something useful in one of these gists:
    # https://gist.github.com/rawiriblundell/2f4712037b2a06155a02a37878ab0c5a
    # https://gist.github.com/rawiriblundell/7e0a302b0ebfe121fedeedb22f521d87
    ...
  ;;
  (*etc etc*)
    ...
  ;;
  (''|*)
    # Then as a catch-all, try feh here
    feh --randomize --bg-fill "$pape"
  ;;
esac
```

Reply:
Yes, that looks like a more elaborate approach. I will work on that. Thanks